Carry #18787: Update Instructions for Docker Remote API access

vdemeester opened the pull request:
Carry #18787 — Using `$DOCKER_CERT_PATH` as the `wget` command below does, I think this is what makes more sense.

🐸

---

Update the location of cert.pem and key.pem to use within the
~/.docker folder
Resolves: #18778

Signed-off-by: Aditi Rajagopal arajagopal@us.ibm.com

thaJeztah commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -64,12 +64,20 @@ without protocol. Throughout this structure, double quotes are required.
+
+wget --no-check-certificate --certificate=$DOCKER_CERT_PATH/cert.pem \
+     --private-key=$DOCKER_CERT_PATH/key.pem \
+     https://your_vm_ip:2376/images/json -O - -q
```
think `your_vm_ip` should be in uppercase here as well. It's a pity that docker-machine doesn't set an env-var containing just the IP-address

vdemeester replied:
I was even wondering if we should put `$(docker-machine ip MACHINE_NAME)`.

thaJeztah commented:
One nit, but LGTM otherwise

MHBauer commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -64,12 +64,20 @@ without protocol. Throughout this structure, double quotes are required.
 
-If you are using `docker-machine`, the Docker daemon is on a virtual host that uses an encrypted TCP socket. This means, for Docker Machine users, you need to add extra parameters to `curl` or `wget` when making test API requests, for example:
+If you are using `docker-machine`, the Docker daemon is on a virtual
+host that uses an encrypted TCP socket. This means, for Docker Machine
```
I'm not sure the 'virtual host' part is *strictly* true (cf. the generic driver), but I don't think it makes any crucial difference in the following use of `curl` or `wget` besides `YOUR_VM`.

vdemeester replied:
hum true, it could be on a cloud provider :-) I could rephrade it to remove the 'virtual part' :wink:

thaJeztah replied:
I also think the text should've been "virtual machine" instead of "virtual host". Agreed on removing "virtual", or simply "the Docker daemon uses an encrypted .."

MHBauer replied:
Heck, we're in the `using docker machine` section, you don't even need anything before the first comma. Also, given all the flags are called `tls`-something, can we work that in there instead of `encrypted tcp`? I don't want to be rewriting everything, but let's be both consistent and specific if possible.

thaJeztah commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -64,12 +64,20 @@ without protocol. Throughout this structure, double quotes are required.
 ## Using Docker Machine with the API
 
-If you are using `docker-machine`, the Docker daemon is on a virtual host that uses an encrypted TCP socket. This means, for Docker Machine users, you need to add extra parameters to `curl` or `wget` when making test API requests, for example:
+If you are using `docker-machine`, the Docker daemon is on host that
```
s/on host/on a host/

Commit:
Update Instructions for Docker Remote API access

Update the location of cert.pem and key.pem to use within the
~/.docker folder
Resolves: #18778

Signed-off-by: Aditi Rajagopal <arajagopal@us.ibm.com>

thaJeztah commented:
LGTM

ping @MHBauer ptal

icecrime commented:
Ping @moxiegirl @MHBauer!

MHBauer commented:
LGTM.
